### ISSUE TITLE
Add tests for benchmarking speed of ColorEmojiFont

### ISSUE THREAD

**Issue by TsundereBug** [TsundereBug/Scolor#16]
```
ColorEmojiFont(
  UShort(32),
  "test",
  Map(
    UInt("A".charAt(0).toInt) -> ColorEmojiEntry(
      f.createDocument("https://dev.w3.org/SVG/tools/svgweb/samples/svg-files/acid.svg"),
      Map()
    )
  )
)
```

@EdgeCaseBerg since you know your way around ScalaTest, is it possible?

**Comment by EdgeCaseBerg** [comment]
@TsundereBug possible to do benchmarks? Definitely. 
If it's something really simple you're wanting we can do this just with some good use of tests + functions like:

```
def timeIt[A](f: => A): (A, Long) = {
  val start = System.nanoTime()
  val result = f
  (result, System.nanoTime() - start)
}
...

val (thing, timeTaken) = timeIt {
   ColorEmojiFont(
     UShort(32),
     "test",
     Map(
        UInt("A".charAt(0).toInt) -> ColorEmojiEntry(
          f.createDocument("https://dev.w3.org/SVG/tools/svgweb/samples/svg-files/acid.svg"),
         Map()
       )
     )
   )
}

```

can do a reasonable and simple speed test if you're just doing it one. And you can of course repeat it a certain number of iterations if you want an average.

```
def timeIt[A](times: Int)(f: => A): Long = {
   val allTimeTaken = (0 to times).map { _ =>
      timeIt(f)._2 // Get just the time it took.
   }.sum
   allTimeTaken / times
}
```

And it's pretty easy to setup a test that prints things out for your information purposes. 

```
"ColorEmojiFont" should "be benchmarked" in {
    val timeTaken = timeIt(10000)(...)
    info(s"It took an average of ${timeTaken} nanoseconds to construct the emoji font")
}
```

All that said, I know that performance testing has tons of little things, like how warmed up the JVM is, what kind of options are enabled, and also stuff related to your machine vs my machine vs a travis build etc. So something like [scalameter](https://scalameter.github.io/) might be good to look into. I read through the getting started and it looks reasonably simple to setup/add. I'm assuming that you can have both scalatest and scalameter as test frameworks side by side but haven't tried it yet.  

If you that code snippet you have is what you're wanting to test, I could probably take a crack at setting up a scalameter test in a few hours when I'm home if you want

**Comment by TsundereBug** [comment]
That sounds amazing. I just want to make sure construction of an emoji font isn't unreasonably slow.

**Comment by EdgeCaseBerg** [comment]
Sounds good, I'll be home sometime within an hour or so and then I should be able to sit down to take a peak at this.

**Comment by EdgeCaseBerg** [comment]
Hey @TsundereBug  I've got something working locally, just going to do one or two tweaks and then I'll send out a PR

**Comment by TsundereBug** [comment]
Fixed by #17
